Assets/InventoryApp.cs

```diff
@@ -0,0 +1,112 @@
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro; // Include the TextMesh Pro namespace
+using System;
+
+public class InventoryApp : MonoBehaviour
+{
+    public Button obtainPotionButton;
+    public Button usePotionButton;  // Button to use a potion, assign in Unity Editor
+    public TMP_Text potionCountText;
+    public TMP_Text cooldownText;
+
+    private int potionCount;
+    private float cooldownTime = 20f;
+    private float nextUseTime;
+
+    void Start()
+    {
+        potionCount = PlayerPrefs.GetInt("HealthPotions", 0);
+
+        // Load saved DateTime for next use
+        string nextUseTimeTicksStr = PlayerPrefs.GetString("NextUseTimeTicks", "0");
+        long nextUseTimeTicks = long.Parse(nextUseTimeTicksStr);
+        DateTime nextUseDateTime = new DateTime(nextUseTimeTicks);
+        
+        // Calculate remaining cooldown in seconds
+        double remainingCooldown = (nextUseDateTime - DateTime.UtcNow).TotalSeconds;
+
+        // Initialize nextUseTime based on remaining cooldown
+        nextUseTime = Time.time + (float)remainingCooldown;
```

Comment:
There is a lot of conversions going on to get the `nextUseTime` :D
- `string`
- `long`
- `DateTime`
- `double`
- `float`

This can definitely be simplified e.g. by using `DateTime.ToString()` and `DateTime.Parse()`